assets/js/googlesitekit/datastore/modules/modules.js

```diff
@@ -0,0 +1,434 @@
+		return API.set( 'core', 'modules', 'activation', { active, slug } );
+	},
+	[ FETCH_MODULES ]: () => {
+		return API.get( 'core', 'modules', 'list', null, { useCache: false } );
+	},
+};
+
+export const reducer = ( state, { type, payload } ) => {
+	switch ( type ) {
+		case START_FETCH_SET_MODULE_STATUS: {
+			const { slug } = payload;
+
+			return {
+				...state,
+				isFetchingSetModuleStatus: {
+					...state.isFetchingSetModuleStatus,
+					[ slug ]: true,
+				},
+			};
+		}
+
+		case FINISH_FETCH_SET_MODULE_STATUS: {
+			const { slug } = payload;
+
+			return {
+				...state,
+				isFetchingSetModuleStatus: {
+					...state.isFetchingSetModuleStatus,
+					[ slug ]: false,
+				},
+			};
+		}
+
+		case CATCH_FETCH_SET_MODULE_STATUS: {
+			const { slug } = payload;
+
+			return {
+				...state,
+				error: payload.error,
+				isFetchingSetModuleStatus: {
+					...state.isFetchingSetModuleStatus,
+					[ slug ]: false,
+				},
+			};
+		}
+
+		case START_FETCH_MODULES: {
+			return {
+				...state,
+				isFetchingModules: true,
+			};
+		}
+
+		case RECEIVE_MODULES: {
+			const { modules } = payload;
+
+			return {
+				...state,
+				modules: modules.reduce( ( acc, module ) => {
+					return { ...acc, [ module.slug ]: module };
+				}, {} ),
+			};
+		}
+
+		case FINISH_FETCH_MODULES: {
+			return {
+				...state,
+				isFetchingModules: false,
+			};
+		}
+
+		case CATCH_FETCH_MODULES: {
+			return {
+				...state,
+				error: payload.error,
+				isFetchingModules: false,
+			};
+		}
+
+		default: {
+			return { ...state };
+		}
+	}
+};
+
+export const resolvers = {
+	*getModules() {
+		const registry = yield Data.commonActions.getRegistry();
+
+		const existingModules = registry.select( STORE_NAME ).getModules();
+
+		if ( ! existingModules ) {
+			yield actions.fetchModules();
+		}
+	},
+};
+
+export const selectors = {
+	/**
+	 * Get the list of modules registered for use with Site Kit.
+	 *
+	 * A module is a section of Site Kit that relates to a particular service,
+	 * like Google Analytics or Google PageSpeed modules. They can provide
+	 * admin-only features (like PageSpeed Insights), frontend-only features,
+	 * or both (eg. Analytics, which can install Analytics <script> tags in the
+	 * frontend, and show dashboards in the WordPress Admin).
+	 *
+	 * Returns a hash of objects with the shape, keyed by slug, when successful:
+	 * ```
+	 * slug: {
+	 *   "slug": "tagmanager",
+	 *   "name": "Tag Manager",
+	 *   "description": "Tag Manager creates an easy to manage way to create tags on your site without updating code.",
+	 *   "homepage": "https://tagmanager.google.com/",
+	 *   "internal": false,
+	 *   "active": false,
+	 *   "connected": false,
+	 *   "dependencies": [
+	 *     "analytics"
+	 *   ],
+	 *   "dependants": []
+	 * }
+	 * ```
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {Object|undefined} Modules available on the site.
```

Comment:
Shouldn't this be `?Object`?

Reply:
`state.modules` has the potential to be `undefined`, so this is correct (though as mentioned in #1468, it should be wrapped in parens).